### DISCUSSION TITLE
Lathe Encoder Ratio / Scale

### DISCUSSION THREAD

**Discussion by raenji-sk** [grblHAL/core#377] [General]
Is there a way to set encoder ratio / scale in a lathe application in grblHAL?
I am trying to convert a lathe to CNC, looking at various controllers but as I still have Phils Teensy board in hand I'll probably start with that. Not sure where or how I will mount the spindle encoder but so far it seems it will have different RPM than spindle so I'd need to set the ratio somewhere. Is there some paramater which can do this? I tried searching but did not find it. Thanks.

**Comment by terjeio** [comment]
It is in the [Wiki](https://github.com/grblHAL/core/wiki/Spindle-sync). FYI Phils T41U5XBB board needs to be modified, @phil-barrett documented the changes somewhere but I was not able to find it again..

**Comment by phil-barrett** [comment]
The [spindle sync conversion thread](https://github.com/terjeio/grblHAL/discussions/289) is in the old terje/grblHAL repo discussions section.

**Comment by raenji-sk** [comment]
Thanks guys, I found those links before. Modding the board to accept the encoder input is no problem. However what I am not sure about is how does it handle different than 1:1 ratios between spindle shaft RPM and encoder shaft RPM. For example encoder has small pulley and does 1 full revolution while spindle has big pulley and does only lets say 0.8 revolutions in the same time.  Do I set the sipndle PPR to manually claculated value (like encoder has 100 ppr, do I set the $38=80 instead of 100?) The Wiki page only has parameter for encoder PPM which in this case would be incorrect no? I looked at Mach3 and LinuxCNC and both have some way of setting the encoder scale or pulley ratios. Hence the original question. Or maybe I am just thikning about it wrong...

**Reply by terjeio** [reply]
$38 is encoder pulses per spindle revolution.
If the index pulse is once per encoder revolution then the code will not work, it has to be once per spindle revolution.,
